templates/named.conf.erb

```diff
@@ -3,11 +3,13 @@
 include "<%= scope.lookupvar('::dns::rndckeypath') %>";
 
 controls  {
-	inet 127.0.0.1 port 953 allow { 127.0.0.1; } keys { "rndc-key"; };
+<%- scope.lookupvar('::dns::controls').sort_by {|k, v| k}.each do |control_ip, control_hash| -%>
+        inet <%= control_ip %> port <%= control_hash['port'] %> allow { <% control_hash['allowed_addresses'].sort.each do |address| %>"<%= address %>"; <% end %>} keys { <% control_hash['keys'].sort.each do |key| %>"<%= key %>"; <% end %>};
```

Comment:
The IP is now quoted where it wasn't before. I think that turns it into an ACL name instead of an IP.